src/proposals/20230802/SparkEthereum_20230802.t.sol

```diff
@@ -0,0 +1,285 @@
+		});
+
+	SparkEthereum_20230802 public payload;
+
+	function setUp() public {
+		vm.createSelectFork(getChain('mainnet').rpcUrl, 17_740_300);
+
+		_selectPayloadExecutor(EXECUTOR);
+
+		payload = new SparkEthereum_20230802();
+	}
+
+	function testSpellExecution() public {
+		IDaiInterestRateStrategy daiStrategy = IDaiInterestRateStrategy(
+			DAI_INTEREST_RATE_STRATEGY_OLD
+		);
+
+		ReserveConfig[] memory allConfigsBefore = createConfigurationSnapshot(
+			'pre-Spark-Ethereum-20230802',
+			POOL
+		);
+
+		/********************************************/
+		/*** Dai Strategy Before State Assertions ***/
+		/********************************************/
+
+		_validateDaiInterestRateStrategy(
+            _findReserveConfigBySymbol(allConfigsBefore, 'DAI').interestRateStrategy,
+            DAI_INTEREST_RATE_STRATEGY_OLD,
+            DaiInterestStrategyValues({
+                vat:                MCD_VAT,
+                pot:                MCD_POT,
+                ilk:                SPARK_ILK,
+                baseRateConversion: RAY,
+                borrowSpread:       0,
+                supplySpread:       0,
+                maxRate:            0.75e27,
+                performanceBonus:   0
+            })
+        );
+
+		daiStrategy.recompute();
+
+		uint256 startingDsr           = IPotLike(MCD_POT).dsr();
+		uint256 startingAnnualizedDsr = _getAnnualizedDsr(startingDsr);
+
+		// ETH-C rate at ~3.14% (currently equals annualized DSR)
+		uint256 stabilityFee = 0.031401763155165655148976000e27;
+
+		( ,, uint256 borrowRate ) = daiStrategy.calculateInterestRates(rateParams);
+
+		assertEq(startingDsr,               1.000000000995743377573746041e27);
+		assertEq(daiStrategy.getBaseRate(), stabilityFee);
+		assertEq(daiStrategy.getBaseRate(), startingAnnualizedDsr);
+		assertEq(borrowRate,                startingAnnualizedDsr);
+
+		uint256 updatedDsr = 1.000000001585489599188229325e27;  // ~5% annualized
+
+		IPotLike(MCD_POT).drip();
+		vm.prank(PAUSE_PROXY);
+		IPotLike(MCD_POT).file('dsr', updatedDsr);
+
+		daiStrategy.recompute();
+
+		uint256 updatedAnnualizedDsr = _getAnnualizedDsr(updatedDsr);
+
+		( ,, borrowRate ) = daiStrategy.calculateInterestRates(rateParams);
+
+		// Demonstrate that old strategy is directly affected by DSR change
+		assertEq(IPotLike(MCD_POT).dsr(),   updatedDsr);
+		assertEq(daiStrategy.getBaseRate(), 0.049999999999999999993200000e27);  // ~5%
+		assertEq(daiStrategy.getBaseRate(), updatedAnnualizedDsr);
+		assertEq(borrowRate,                updatedAnnualizedDsr);
+
+
+		// Go back to starting state before execution
+		IPotLike(MCD_POT).drip();
+		vm.prank(PAUSE_PROXY);
+		IPotLike(MCD_POT).file('dsr', startingDsr);
+
+		daiStrategy.recompute();
+
+		( ,, borrowRate ) = daiStrategy.calculateInterestRates(rateParams);
+
+		assertEq(daiStrategy.getBaseRate(), startingAnnualizedDsr);  // Back to 3.14%
+		assertEq(borrowRate,                startingAnnualizedDsr);
```

Comment:
All this extra testing should be moved to a separate test. `testSpellExecution` should just test and diff the spell with the collateral assertions at the bottom. The problem is by doing all these extra functions (sometimes state-modifying it may make us miss something). For example the diff is showing oracle price changes on sDAI which is not relevant when we just want to see the isolated state change for the spell execution alone.

Reply:
Updated